packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/Tasks.spec.js

```diff
@@ -1,133 +1,170 @@
+      step = rootStore.workflowSteps.active
+    })
 
-  it('should render the correct task component if the workflow is loaded', function () {
-    const wrapper = shallow(
-      <Tasks
-        loadingState={asyncStates.success}
-        ready
-        addAnnotation={addAnnotation}
-        classification={classification}
-        tasks={tasks}
-      />
-    )
-    // Is there a better way to do this?
-    expect(wrapper.find('SingleChoiceTask')).to.have.lengthOf(1)
-  })
+    describe(`Task ${taskType}`, function () {
+      it('should render without crashing', function () {
+        const wrapper = shallow(<Tasks />)
+        expect(wrapper).to.be.ok()
+      })
 
-  describe('task components', function () {
-    let taskWrapper
+      it('should render null on initialization', function () {
+        const wrapper = shallow(<Tasks />)
+        expect(wrapper.type()).to.be.null()
+      })
 
-    describe('while the subject is loading', function () {
-      before(function () {
-        const wrapper = shallow(
-          <Tasks
-            addAnnotation={addAnnotation}
-            classification={classification}
-            loadingState={asyncStates.success}
-            subjectReadyState={asyncStates.loading}
-            tasks={tasks}
-          />
-        )
-        taskWrapper = wrapper.find('SingleChoiceTask')
+      it('should render a loading UI when the workflow loading', function () {
+        const wrapper = shallow(<Tasks loadingState={asyncStates.loading} />)
+        expect(wrapper.contains('Loading')).to.be.true()
       })
 
-      it('should be disabled', function () {
-        expect(taskWrapper.prop('disabled')).to.be.true()
+      it('should render an error message when there is a loading error', function () {
+        const wrapper = shallow(<Tasks loadingState={asyncStates.error} />)
+        expect(wrapper.contains('Something went wrong')).to.be.true()
       })
-    })
 
-    describe('when the subject viewer is ready', function () {
-      before(function () {
+      it('should render null if the workflow is load but has no tasks', function () {
+        const wrapper = shallow(<Tasks loadingState={asyncStates.success} ready />)
+        expect(wrapper.type()).to.be.null()
+      })
+
+      it('should render the correct task component if the workflow is loaded', function () {
         const wrapper = shallow(
           <Tasks
+            loadingState={asyncStates.success}
+            ready
             addAnnotation={addAnnotation}
             classification={classification}
-            loadingState={asyncStates.success}
-            subjectReadyState={asyncStates.success}
-            step={step}
-            tasks={tasks}
+            tasks={step.tasks}
           />
         )
-        taskWrapper = wrapper.find('SingleChoiceTask')
+        // Is there a better way to do this?
+        expect(wrapper.find(TaskComponent.displayName)).to.have.lengthOf(1)
```

Comment:
```js
wrapper.find(TaskComponent)
```
returns 0 nodes, so I've fallen back on searching by display name. I can't see why searching for the component itself would fail.

Reply:
FWIW, my WIP Grommet upgrade branch is updating Enzyme use to take advantage of the new options available for injecting context. I'm not sure if there have been any improvements in the minor version update for this.